vllm/model_executor/layers/linear.py

```diff
@@ -260,9 +264,13 @@ def weight_loader(self,
                       param: Parameter,
                       loaded_weight: torch.Tensor,
                       loaded_shard_id: Optional[int] = None):
-        param_data = param.data
+        param_data = get_param_data(param)
         output_dim = getattr(param, "output_dim", None)
         if loaded_shard_id is None:
+            if isinstance(param, SparseParameter):
```

Comment:
What's the reason for this limitation?

Reply:
@alexm-nm - we should be able to support this. But I don't have / know of an example of a model yet which has this so I wasn't sure of how to test this

Reply:
Ok cool